lib/internal/modules/esm/get_format.js

```diff
@@ -46,13 +55,16 @@ function defaultGetFormat(url, context, defaultGetFormatUnused) {
       /^([^/]+\/[^;,]+)(?:[^,]*?)(;base64)?,/,
       parsed.pathname,
     ) || [ null, null, null ];
-    const format = ({
-      '__proto__': null,
-      'text/javascript': 'module',
-      'application/json': experimentalJsonModules ? 'json' : null,
-      'application/wasm': experimentalWasmModules ? 'wasm' : null
-    })[mime] || null;
-    return { format };
+    return { format: mimeToFormat(mime) };
+  } else if (experimentalHttpsModules && (
+    parsed.protocol === 'https:' ||
+    parsed.protocol === 'http:'
+  )) {
+    return fetch(parsed).then((res) => {
```

Comment:
```suggestion
    return PromisePrototypeThen(fetch(parsed), (res) => {
```
alternatively this could use await